cell_workers/pokemon_catch_worker.py

```diff
@@ -105,7 +106,24 @@ def work(self):
                                 if status is 3:
                                     print('[x] Oh no! ' + str(pokemon_name) + ' vanished! :(')
                                 if status is 1:
-                                    if cp < self.config.cp:
+                                    self.api.get_inventory()
+                                    response_dict = self.api.call()
+                                    if self.config.cp == "smart":
+                                        id_cp_tuples = self.get_id_cp_tuples_for_pokemonid(pokemon['pokemon_data']['pokemon_id'],response_dict)
+                                        prev_id, prev_cp = (0,0)
+                                        for id_cp in id_cp_tuples:
+                                            next_id,next_cp = id_cp
+                                            if next_cp < prev_cp:
+                                                self.transfer_pokemon(next_id)
+                                            elif prev_id != 0:
+                                                self.transfer_pokemon(prev_id)
+                                            prev_id,prev_cp = id_cp
+                                    try:
```

Comment:
@mohitmun please add an `else` clause to the `if self.config.cp == "smart"` if and move the `try` block inside it.